impacket/examples/ntlmrelayx/clients/ldaprelayclient.py

```diff
@@ -172,7 +174,14 @@ def __init__(self, serverConfig, target, targetPort = 636, extendedSecurity=True
         LDAPRelayClient.__init__(self, serverConfig, target, targetPort, extendedSecurity)
 
     def initConnection(self):
-        self.server = Server("ldaps://%s:%s" % (self.targetHost, self.targetPort), get_info=ALL)
-        self.session = Connection(self.server, user="a", password="b", authentication=NTLM)
-        self.session.open(False)
-        return True
+        try:
+            tls_config = Tls(ciphers='ALL:@SECLEVEL=0', version=ssl.PROTOCOL_TLSv1_2)
+            self.server = Server("ldaps://%s:%s" % (self.targetHost, self.targetPort), get_info=ALL, tls=tls_config)
+            self.session = Connection(self.server, user="a", password="b", authentication=NTLM)
+            self.session.open(False)
+        except LDAPSocketOpenError:
+            tls_config = Tls(ciphers='ALL:@SECLEVEL=0', version=ssl.PROTOCOL_TLSv1)
+            self.server = Server("ldaps://%s:%s" % (self.targetHost, self.targetPort), get_info=ALL, tls=tls_config)
+            self.session = Connection(self.server, user="a", password="b", authentication=NTLM)
+            self.session.open(False)
+        return True
```

Comment:
Thanks!

Think it's better something like this, to avoid code duplications
```suggestion
        try:
            self._openConnection(ssl.PROTOCOL_TLSv1_2)
        except LDAPSocketOpenError:
            self._openConnection(ssl.PROTOCOL_TLSv1)
        return True

    def _openConnection(self, version):
        tls_config = Tls(ciphers='ALL:@SECLEVEL=0', version=version)
        self.server = Server("ldaps://%s:%s" % (self.targetHost, self.targetPort), get_info=ALL, tls=tls_config)
        self.session = Connection(self.server, user="a", password="b", authentication=NTLM)
        self.session.open(False)
```

(Know the other use cases are duplicating code. We should fix them eventually in other pr)

Will start checking your fix in action. Will let you know how that goes
Thank you!